lando/k8s/lando.py

```diff
@@ -0,0 +1,251 @@
+import logging
+import json
+import sys
+from lando.server.lando import Lando, JobStates, JobSteps, JobSettings, BaseJobActions
+from lando.k8s.cluster import ClusterApi
+from lando.k8s.jobmanager import JobManager
+from lando.k8s.config import create_server_config
+
+
+class K8sJobSettings(JobSettings):
+    def get_cluster_api(self):
+        settings = self.config.cluster_api_settings
+        return ClusterApi(settings.host, settings.token, settings.namespace, verify_ssl=settings.verify_ssl)
+
+
+class K8sJobActions(BaseJobActions):
+    """
+    Used by K8sLando to handle messages at a job specific context.
+    """
+    def __init__(self, settings):
+        super(K8sJobActions, self).__init__(settings)
+        self.cluster_api = settings.get_cluster_api()
+
+    def make_job_manager(self):
+        job = self.job_api.get_job()
+        return JobManager(self.cluster_api, self.settings.config, job)
+
+    def job_is_at_state_and_step(self, state, step):
+        job = self.job_api.get_job()
+        return job.state == state and job.step == step
+
+    def start_job(self, payload):
+        """
+        Request from user to start running a job. This starts a job to stage user input data into a volume.
+        :param payload:StartJobPayload contains job_id we should start
+        """
+        self._set_job_state(JobStates.RUNNING)
+        self._set_job_step(JobSteps.CREATE_VM)
+        manager = self.make_job_manager()
+
+        self._show_status("Creating stage data persistent volumes")
+        manager.create_stage_data_persistent_volumes()
+
+        self.perform_staging_step()
+
+    def perform_staging_step(self):
+        manager = self.make_job_manager()
```

Comment:
I'm not sure I understand the lifecycle of JobManagers. Is there a reason each method creates its own fresh instance? Would anything be lost by creating one in `__init__()` and keeping it for the life of this object? 

I suspect there may be an issue of stale job data from the API. If that's the case, it probably deserves a comment.

Reply:
I moved construction of JobManager and a cached __bespin__ job to `__init__`. We only read state/step from the __bespin__ job to determine what to do. After start changing state/step we don't read this stale data. To ward off future bugs I added changes to keep the cached __bespin__ job up to date.